stubs/PyScreeze/pyscreeze/__init__.pyi

```diff
@@ -0,0 +1,196 @@
+    grayscale: bool | None = ...,
+    limit: _Unused = ...,
+    region: tuple[int, int, int, int] | None = ...,
+    step: int = ...,
+    confidence: SupportsFloat | SupportsIndex | str = ...,
+) -> Box | None: ...
+
+# _locateAll_python / _locateAll_pillow
+@overload
+def locateOnScreen(
+    image: str | Image.Image,
+    minSearchTime: float = ...,
+    *,
+    grayscale: bool | None = ...,
+    limit: _Unused = ...,
+    region: tuple[int, int, int, int] | None = ...,
+    step: int = ...,
+    confidence: None = ...,
+) -> Box | None: ...
+
+# _locateAll_opencv
+@overload
+def locateAllOnScreen(
+    image: str | Image.Image | _Mat,
+    *,
+    grayscale: bool | None = ...,
+    limit: int = ...,
+    region: tuple[int, int, int, int] | None = ...,
+    step: int = ...,
+    confidence: SupportsFloat | SupportsIndex | str = ...,
+) -> Generator[Box, None, None]: ...
+
+# _locateAll_python / _locateAll_pillow
+@overload
+def locateAllOnScreen(
+    image: str | Image.Image,
+    *,
+    grayscale: bool | None = ...,
+    limit: int | None = ...,
+    region: tuple[int, int, int, int] | None = ...,
+    step: int = ...,
+    confidence: None = ...,
+) -> Generator[Box, None, None]: ...
+
+# _locateAll_opencv
+@overload
+def locateCenterOnScreen(
+    image: str | Image.Image | _Mat,
+    *,
+    minSearchTime: float,
+    grayscale: bool | None = ...,
+    limit: _Unused = ...,
+    region: tuple[int, int, int, int] | None = ...,
+    step: int = ...,
+    confidence: SupportsFloat | SupportsIndex | str = ...,
+) -> Point | None: ...
+
+# _locateAll_python / _locateAll_pillow
+@overload
+def locateCenterOnScreen(
+    image: str | Image.Image,
+    *,
+    minSearchTime: float,
+    grayscale: bool | None = ...,
+    limit: _Unused = ...,
+    region: tuple[int, int, int, int] | None = ...,
+    step: int = ...,
+    confidence: None = ...,
+) -> Point | None: ...
+
+# _locateAll_opencv
+@overload
+def locateOnWindow(
+    image: str | Image.Image | _Mat,
+    title: str,
+    *,
+    grayscale: bool | None = ...,
+    limit: _Unused = ...,
+    step: int = ...,
+    confidence: SupportsFloat | SupportsIndex | str = ...,
+) -> Box | None: ...
+
+# _locateAll_python / _locateAll_pillow
+@overload
+def locateOnWindow(
+    image: str | Image.Image,
+    title: str,
+    *,
+    grayscale: bool | None = ...,
+    limit: _Unused = ...,
+    step: int = ...,
+    confidence: None = ...,
+) -> Box | None: ...
+def showRegionOnScreen(region: tuple[int, int, int, int], outlineColor: str = ..., filename: str = ...) -> None: ...
+def center(coords: Box) -> Point: ...
+def pixelMatchesColor(
+    x: int, y: int, expectedRGBColor: tuple[int, int, int] | tuple[int, int, int, int], tolerance: int = ...
+) -> bool: ...
+def pixel(x: int, y: int) -> tuple[int, int, int]: ...
+def screenshot(imageFilename: StrOrBytesPath | None = ..., region: tuple[int, int, int, int] | None = ...) -> Image.Image: ...
+
+grab = screenshot
+# _locateAll_opencv
+@overload
```

Comment:
This is a strange use of overloads, since it looks like either one variant or the other is used depending on whether cv2 is present. Overloads are normally used when there are complicated relationships between different parameters and/or the return type. On the other hand, I can't think of any concrete ill effect of using overloads here, so I'm fine with keeping them.

Reply:
Yeah, since we can't represent that condition in typing, overloads are the next best thing so that both possible signatures are accurate.